src/main/java/com/slcube/shop/business/review/service/ReviewServiceImpl.java

```diff
@@ -0,0 +1,85 @@
+package com.slcube.shop.business.review.service;
+
+import com.slcube.shop.business.item.domain.Item;
+import com.slcube.shop.business.item.repository.ItemRepository;
+import com.slcube.shop.business.item.repository.ItemRepositoryHelper;
+import com.slcube.shop.business.review.domain.ReportedReview;
+import com.slcube.shop.business.review.domain.Review;
+import com.slcube.shop.business.review.dto.*;
+import com.slcube.shop.business.review.repository.ReportedReviewRepository;
+import com.slcube.shop.business.review.repository.ReviewRepository;
+import com.slcube.shop.business.review.repository.ReviewRepositoryHelper;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional(readOnly = true)
+@RequiredArgsConstructor
+public class ReviewServiceImpl implements ReviewService {
+
+    private final ReviewRepository reviewRepository;
+    private final ReviewRepositoryHelper reviewRepositoryHelper;
+    private final ItemRepository itemRepository;
+    private final ItemRepositoryHelper itemRepositoryHelper;
+    private final ReportedReviewRepository reportedReviewRepository;
+
+    @Override
+    public Long saveReview(ReviewSaveRequestDto requestDto) {
+        Long itemId = requestDto.getItemId();
+        Item item = itemRepositoryHelper.findByNotDeleted(itemRepository, itemId);
```

Comment:
필드 변수가 너무 많아지고 있네요 나중에 mocking할때 힘들 수 있을거같아요
```suggestion
    private final ReviewRepository reviewRepository;
    private final ReviewRepositoryHelper reviewRepositoryHelper;
    private final ReviewValidation reviewValidation;
    private final ReportedReviewRepository reportedReviewRepository;

    @Override
    public Long saveReview(ReviewSaveRequestDto requestDto) {
        Long itemId = requestDto.getItemId();
        Item item = reviewValidation.validateCreateReview(itemId);
```

요렇게 하는게 어떨까요?
그러면 필드변수 2개를 줄일 수 있을거같아요

Reply:
validation은 편하게 바꾸실껀지 아니면 안바꾸실껀지 결정해서 반영 해주세요~!